src/hooks/use-show-value-outside.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useRef } from "react"
+import { isEmpty } from "ramda"
+
+import { ChartData, DygraphData } from "domains/chart/chart-types"
+import { Attributes } from "domains/chart/utils/transformDataAttributes"
+import { ChartLibraryConfig } from "domains/chart/utils/chartLibrariesSettings"
+
+
+interface UseShowValueOutsideArgument {
+  attributes: Attributes
+  chartData: ChartData
+  chartSettings: ChartLibraryConfig
+  hoveredRow: number
+  legendFormatValue: ((v: number | string | null) => number | string)
+  showUndefined: boolean
+}
+
+// example of the attribute:
+// show-value-of-iowait-at: "system.cpu.iowait.1"
+
+export const useShowValueOutside = ({
+  attributes,
+  chartData,
+  chartSettings,
+  hoveredRow,
+  legendFormatValue,
+  showUndefined,
+}: UseShowValueOutsideArgument) => {
+  // a ref to store found elements, just once per lifetime of component
+  const showValueAttributesNodes = useRef<(HTMLElement | null)[]>([])
+
+  // find the nodes that will have populated values
+  useEffect(() => {
+    const { showValueOf } = attributes
+    // showValueOf will be undefined if not used, but additional isEmpty check can prevent
+    // regression performance issue in the future
+    if (!showValueOf || isEmpty(showValueOf)) {
+      return
+    }
+    const dimensionNames = chartData.dimension_names
+    const dimensionIds = chartData.dimension_ids
+    dimensionNames.forEach((dimensionName, i) => {
+      const userElementId = showValueOf[`show-value-of-${dimensionName.toLowerCase()}`]
+        || showValueOf[`show-value-of-${dimensionIds[i].toLowerCase()}-at`]
+
+      // if element is not found, just add null
+      showValueAttributesNodes.current = showValueAttributesNodes.current.concat(
+        document.getElementById(userElementId),
+      )
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []) // do only once on purpose
```

Comment:
I think, react-use has an effect for this, which doesn't require eslint disable / additional commenting. Like useDidMount or something.

Reply:
good idea, will do that now